src/pmdas/openvswitch/pmdaopenvswitch.py

```diff
@@ -0,0 +1,184 @@
+            [ 'bridge.rstp_status',                                                     PM_TYPE_STRING, PM_SEM_COUNTER, pmUnits(),  'rapid spanning tree protocol status'], # 19
+            [ 'bridge.sflow',                                                           PM_TYPE_STRING, PM_SEM_COUNTER, pmUnits(),  'sFlow is a sampling technology that meets the key requirements for a network traffic monitoring solution'], # 20
+            [ 'bridge.status',                                                          PM_TYPE_STRING, PM_SEM_COUNTER, pmUnits(),  'status'], # 21
+            [ 'bridge.stp_enable',                                                      PM_TYPE_STRING, PM_SEM_COUNTER, pmUnits(),  'spanning tree protocol'] # 22
+        ]
+
+        for item in range(len(self.bridge_metrics)):
+            self.add_metric(name + '.' +
+                            self.bridge_metrics[item][0],
+                            pmdaMetric(self.pmid(self.bridge_cluster, item),
+                                       self.bridge_metrics[item][1],
+                                       self.bridge_indom,
+                                       self.bridge_metrics[item][2],
+                                       self.bridge_metrics[item][3]),
+                            self.bridge_metrics[item][4],
+                            self.bridge_metrics[item][4])
+
+        self.set_fetch_callback(self.openvswitch_fetch_callback)
+        self.set_refresh(self.openvswitch_refresh)
+        self.set_user(PCP.pmGetConfig('PCP_USER'))
+
+    def fetch_bridge_info(self):
+        """ fetches result from command line """
+
+        query = ['sudo', 'ovs-vsctl', '--format=json', 'list', 'bridge']
+        out = subprocess.Popen(query, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        stdout, stderr = out.communicate()
+        stdout = stdout.decode("utf-8")
+        return stdout, stderr
+
+    def get_bridge_info_json(self):
+        """ Convert the commandline output to json """
+
+        stdout, stderr = self.fetch_bridge_info()
+
+        if stderr is None:
+            temp = json.loads(stdout)
+            # reorganize json a bit
+            self.bridge_info_json = dict()
+            self.bridge_names = []
+            for idx in range(len(temp)):
+                self.bridge_info_json[str(temp["data"][idx][13])] = temp["data"][idx]
+                self.bridge_names.append(str(temp["data"][idx][13]))
+
+    def bridge_instances(self):
+        """ set names for openvswitch instances """
+        insts = []
+        for idx, val in enumerate(self.bridge_names):
+            insts.append(pmdaInstid(idx, val))
+        self.add_indom(pmdaIndom(self.bridge_indom, insts))
+
+    def openvswitch_refresh(self, cluster):
+        """refresh function"""
+        if cluster == self.bridge_cluster:
+            # self.get_bridge_info_json()
+            insts = []
```

Comment:
As of now just facing a bit of a hiccup here; Gives out an unknown error while uncommenting this.